packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -117,6 +117,14 @@ export interface EngineReportingOptions<TContext> {
    * TODO(helen): LINK TO EXAMPLE FUNCTION? e.g. a function recursively search for keys to be blocklisted
    */
   sendVariableValues?: VariableValueOptions;
+  /**
+   * By default Apollo server will report all queries to Graph Manager.
+   * If you would like to disable reporting and tracing for certain queries you can give a function that takes in
+   * graphqlRequestContext. The return value of the function will be used to determine if any stats or tracing information will be collected for the query.
+   * Always returning false will behave similarly to turning off reporting and no instrumentation will take place.
+   *
+   */
+  shouldReportQuery?: (request: GraphQLRequestContext) => boolean;
```

Comment:
How about just `report`? Particularly because:

* `Query` is still the umbrella term for "operation" that we're trying to (slowly) move away from as a blanket term; and
* I think to be consistent we'd need to add `should` to other predicate things across the Apollo Server stack to have a consistent public API (e.g., maybe not this specific example but: even `sendVariableValues` above this might need to be `shouldSendVariableValues`).

Reply:
Went with `traceReporting` to mirror `experimental_schemaReporting`